File: frontend/src/shared/types/modules.ts
import type { ModuleId } from "@/shared/types/common";
import type { ActiveFlag } from "@/shared/types/dashboard";

export interface ModuleDefinition {
  module_id: ModuleId;
  module_name: string;
  description: string;
}

export interface ModulesResponse {
  modules: ModuleDefinition[];
}

export interface ModuleSignal {
  date: string;
  module_id: ModuleId;
  signal_name: string;
  raw_value: number;
  mad_score: number;
  flag: boolean;
  unit: string;
}

export interface ModuleSnapshot {
  module_id: ModuleId;
  module_name: string;
  module_score?: number;
  signals: ModuleSignal[];
  active_flags: ActiveFlag[];
}

export interface ModulesSnapshotResponse {
  date: string;
  modules: ModuleSnapshot[];
}

export interface ModuleSignalsResponse {
  module_id: ModuleId;
  signals: ModuleSignal[];
  active_flags: ActiveFlag[];
}

export interface ModuleSignalsParams {
  from?: string;
  to?: string;
}

File: backend/internal/dto/modules.go
package dto

type ModuleDefinition struct {
	ModuleID    string `json:"module_id"`
	ModuleName  string `json:"module_name"`
	Description string `json:"description"`
}

type ModulesListResponse struct {
	Modules []ModuleDefinition `json:"modules"`
}

type ModuleSignal struct {
	Date       string  `json:"date"`
	ModuleID   string  `json:"module_id"`
	SignalName string  `json:"signal_name"`
	RawValue   float64 `json:"raw_value"`
	MadScore   float64 `json:"mad_score"`
	Flag       bool    `json:"flag"`
	Unit       string  `json:"unit"`
}

type ModuleSignalsResponse struct {
	ModuleID    string         `json:"module_id"`
	Signals     []ModuleSignal `json:"signals"`
	ActiveFlags []ActiveFlag   `json:"active_flags"`
}

type ModuleSnapshot struct {
	ModuleID    string         `json:"module_id"`
	ModuleName  string         `json:"module_name"`
	ModuleScore float64        `json:"module_score"`
	Signals     []ModuleSignal `json:"signals"`
	ActiveFlags []ActiveFlag   `json:"active_flags"`
}

type ModulesSnapshotResponse struct {
	Date    string           `json:"date"`
	Modules []ModuleSnapshot `json:"modules"`
}
